### PR TITLE
Remove other uses of /Zc:__cplusplus in the documentation.

### PR DESCRIPTION
**Pull request by rcurtin**
This is just a follow-up to #3555, since that option is no longer required for MSVC.

### CODE REVIEW

**Comment by conradsnicta** [comment]
There's a small fly in the ointment.  Perhaps merge this only just before (or just after) the release of mlpack 4.2.2.

mlpack 4.2.1 does not have the updated C++17 detection under MSVC.  If people download mlpack 4.2.1 (from mlpack.org) and then follow the updated instructions (doc/user/build_windows.md and doc/user/sample_ml_app.md), mlpack 4.2.1 is not going to work for them under MSVC.

Right now the current documentation works for both mlpack 4.2.1 and the git master branch.  Use of the _MSVC_LANG macro (in #3555) can be thought of as a workaround for the bizarre and confusing config options to enable C++17 under MSVC.  Suggest to keep the current documentation to cover the changeover period between mlpack 4.2.1 and 4.2.2.

**Comment by rcurtin** [comment]
Yes, true, although the documentation distributed with 4.2.1 will be accurate. (It is only inaccurate if a user comes to Github to read documentation, which is probably the most common use case realistically like you pointed out.) I don't mind releasing 4.2.2 immediately after merging this, it's easy enough and minimizes the period of inaccurate documentation for older versions.

**Review by conradsnicta — APPROVED** [review]

**Comment by rcurtin** [comment]
#3547 is so close that I'm going to wait for that to be merged to merge this, and then open a release PR.

**Pull request merged**
